From da7b2d9667b54450482795842c2ac507778e9529 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Tue, 20 Aug 2024 09:37:13 +0300
Subject: [PATCH] [FSW-7256] fs1: declare access zones

With this patch, added enum which declares
various areas which require access during
rom recovery.

Change-Id: Ifdce7cfb1ae064c6279c6a18eff86c8de67b6654
---

diff --git a/arch/fs1/arc/dev_security.h b/arch/fs1/arc/dev_security.h
index 611fe7d..65f4997 100644
--- a/arch/fs1/arc/dev_security.h
+++ b/arch/fs1/arc/dev_security.h
@@ -31,6 +31,15 @@
     CRC_FAIL,
 };
 
+/* Various blocks to be configured during boot rom in recovery */
+enum security_access_ctrl_zones {
+	SEC_ACCESS_CTRL_QSPI,
+	SEC_ACCESS_CTRL_PAD_SEL,
+	SEC_ACCESS_CTRL_PSOC_SEC,
+	SEC_ACCESS_CTRL_GEN_MEM,
+	SEC_ACCESS_CTRL_MAX,
+};
+
 void security_load_revoke_bits(void);
 int security_efuse_image_validation_en(void);
 void boot_rom_eom_overrides(void);
